### ISSUE TITLE
Ensure matching client version when connecting to a repository

### ISSUE THREAD

**Issue by cr0** [syncany/syncany#295]
Syncany doesn't validate if all clients connected to a repository use the same Syncany version. Because of different storage concepts*, this can result in a damaged repository either by accident due to a missed update or on purpose.

**Proposal**
Init writes a version identifier in the repository config. All connecting clients have to have at least this version. If the storage concept changes, the first new client updates the version tag in the config and disconnects all other clients.
We can either use a) `Syncany major.minor` or b) a separate storage concept identifier as a version tag.

**Comment by binwiederhier** [comment]
@cr0 Friendly bump. 
Waffle tells me you're assigned to this and it's in progress :)

**Comment by cr0** [comment]
Yep it's still in the making and will be my task once #300 and #335 are finished.

**Comment by cr0** [comment]
I started on this issue but I don't know how to proceed. Syncany now verifies that the remote repo is supported. However, if the repo is already connected the remote `syncany` file isn't read again during `up` and  `down` so we cannot detect any changes to the version (which currently does not change). Therefore we need to implement a `verifyRepo` method similar to the one in `ConnectOperation`. Should I use `AbstractTransferOperation`? If so, we still have some redundancies since the same method is already implemented in `ConnectOperation` which again does not extend the said class.

Any ideas?
